zeth_contracts/contracts/BaseMerkleTree_test.sol

```diff
@@ -0,0 +1,64 @@
+// Copyright (c) 2015-2020 Clearmatics Technologies Ltd
```

Comment:
If I've understood correctly, this duplicates all of the merkle tree contract code which is a bit concerning.
I think the original problem is that we try to return a set of nodes that is too large.
Could this not be fixed by having `MerkleTreeMiMC7_test::testAddLeaves` return just the new root and compare that to the version on the host?  I think this would still give us a high level of confidence that the contract was doing the right thing.

If we want to check specific nodes on the contract, perhaps `MerkleTreeMiMC7_test` could expose another method to return single nodes, or ranges of nodes, but this is probably not necessary if we check that the root is consistent.

Reply:
Yes I've switched to that. Don't know why the full set of nodes was returned before TBH...
The tests only require the value of `nodes[0]` (the root) to work.
It seems that we were trying to return the full set of tree nodes just to build an error message https://github.com/clearmatics/zeth/blob/develop/client/test_commands/test_merkle_tree_contract.py#L44